test: Add airflow 2.10.5

NickLarsenNZ opened the pull request:
Part of https://github.com/stackabletech/docker-images/issues/1074

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

### Author

- [x] Integration tests passed (for non trivial changes)

### Reviewer

- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)

### Acceptance

- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated

NickLarsenNZ commented:
Tests pass

```
        --- PASS: kuttl/harness/ldap_airflow-latest-2.10.5_ldap-authentication-no-tls_openshift-false_executor-kubernetes (304.14s)
        --- PASS: kuttl/harness/ldap_airflow-latest-2.10.5_ldap-authentication-no-tls_openshift-false_executor-celery (192.61s)
        --- PASS: kuttl/harness/ldap_airflow-latest-2.10.5_ldap-authentication-insecure-tls_openshift-false_executor-kubernetes (151.50s)
        --- PASS: kuttl/harness/ldap_airflow-latest-2.10.5_ldap-authentication-insecure-tls_openshift-false_executor-celery (183.58s)
        --- PASS: kuttl/harness/resources_airflow-latest-2.10.5_openshift-false (138.30s)
        --- PASS: kuttl/harness/opa_airflow-2.10.5_opa-latest-1.4.2_openshift-false (125.27s)
        --- PASS: kuttl/harness/orphaned-resources_airflow-latest-2.10.5_openshift-false (151.51s)
        --- PASS: kuttl/harness/mount-dags-configmap_airflow-latest-2.10.5_openshift-false_executor-kubernetes (154.26s)
        --- PASS: kuttl/harness/mount-dags-configmap_airflow-latest-2.10.5_openshift-false_executor-celery (169.48s)
        --- PASS: kuttl/harness/smoke_airflow-2.10.5_openshift-false_executor-celery (216.36s)
        --- PASS: kuttl/harness/cluster-operation_airflow-latest-2.10.5_openshift-false (257.01s)
        --- PASS: kuttl/harness/mount-dags-gitsync_airflow-latest-2.10.5_openshift-false_executor-kubernetes (234.85s)
        --- PASS: kuttl/harness/mount-dags-gitsync_airflow-latest-2.10.5_openshift-false_executor-celery (232.36s)
        --- PASS: kuttl/harness/external-access_airflow-2.10.5_openshift-false (206.11s)
        --- PASS: kuttl/harness/overrides_airflow-latest-2.10.5_openshift-false (158.00s)
        --- PASS: kuttl/harness/logging_airflow-2.10.5_openshift-false_executor-kubernetes (283.15s)
        --- PASS: kuttl/harness/oidc_airflow-2.10.5_openshift-false (141.56s)
        --- PASS: kuttl/harness/ldap_airflow-latest-2.10.5_ldap-authentication-server-verification-tls_openshift-false_executor-kubernetes (150.10s)
        --- PASS: kuttl/harness/ldap_airflow-latest-2.10.5_ldap-authentication-server-verification-tls_openshift-false_executor-celery (192.91s)
        --- PASS: kuttl/harness/smoke_airflow-2.10.5_openshift-false_executor-kubernetes (187.23s)
        --- PASS: kuttl/harness/logging_airflow-2.10.5_openshift-false_executor-celery (288.98s)
```

lfrancke commented:
Could you also update this page please: https://github.com/stackabletech/airflow-operator/blob/chore/bump-airflow-pre-25.7.0/docs/modules/airflow/pages/required-external-components.adoc

- https://airflow.apache.org/docs/apache-airflow/2.10.5/installation/prerequisites.html

PostgreSQL: Remove 11, add 16
MySQL: Remove 5.7, add "Innovation"
MSSQL has been removed in 2.9
Is it even possible to use SQLite with our operator? If not I'd just remove it too..

We really need a checklist item to update this page for each of our tools.... or link to the upstream page.

NickLarsenNZ commented:
> Could you also update this page please: [`chore`/bump-airflow-pre-25.7.0/docs/modules/airflow/pages/required-external-components.adoc](https://github.com/stackabletech/airflow-operator/blob/chore/bump-airflow-pre-25.7.0/docs/modules/airflow/pages/required-external-components.adoc)
> 
>     * [airflow.apache.org/docs/apache-airflow/2.10.5/installation/prerequisites.html](https://airflow.apache.org/docs/apache-airflow/2.10.5/installation/prerequisites.html)
> 
> 
> PostgreSQL: Remove 11, add 16 MySQL: Remove 5.7, add "Innovation" MSSQL has been removed in 2.9 Is it even possible to use SQLite with our operator? If not I'd just remove it too..
> 
> We really need a checklist item to update this page for each of our tools.... or link to the upstream page.

I have updated the templates for the products that have operators with a `required-external-components.adoc` page:

https://github.com/stackabletech/docker-images/pull/1109

I have manually updated the existing issues to include the task.

Lastly, I have updated the prereqs in 9e706712514dcbd8d37a5a0d0515d8793eb6e745

lfrancke reviewed (approved):
Thank you very much for the additional changes. I didn't test this but it looks good to me!
